src/z_zone.h

```diff
@@ -40,6 +40,7 @@
 
 typedef enum {
   PU_STATIC,
+  PU_VIDEO, // [Nugget] Dynamic arrays
```

Comment:
I don't get it. Why is this necessary?

Reply:
This implementation is loosely based on code from Eternity, which added a `PU_VALLOC` tag to be used for the memory allocated for these arrays. I figured I should add a new tag too, but I must admit I've no idea of what exactly it is for. What other tag do you think I should use?

Reply:
Whenever you use `Z_Malloc()` in a way where you would usually use plain `malloc()` (i.e. without any magic involved to automatically free resources e.g. when finishing a map) that'd be `PU_STATIC`.